compiler/luci/service/src/CircleShapeInferenceRule.cpp

```diff
@@ -1312,8 +1312,13 @@ loco::NodeShape infer_slice(const luci::CircleSlice *node)
 
   auto input_shape = luci::shape_get(node->input()).as<loco::TensorShape>();
 
-  auto const_begin = loco::must_cast<luci::CircleConst *>(node->begin());
-  auto const_size = loco::must_cast<luci::CircleConst *>(node->size());
+  auto const_begin = dynamic_cast<luci::CircleConst *>(node->begin());
+  auto const_size = dynamic_cast<luci::CircleConst *>(node->size());
+  if (const_begin == nullptr || const_size == nullptr)
+  {
+    // NOTE shape depends on values of begin and size
+    return use_own(node);
```

Comment:
This case seems to be dynamic shape.

If we don't know `begin()` or `size()`, we cannot decide each dimension values.
However, there is one thing we can decide, which is rank :D

For example, let's assume following situation.
- `input()`, `begin()`, `size()` are not `CircleConst`
- shape of  `input()` : [1, 2, 3, 4]
- shape of  `begin()` : [4]
- shape of `size()` : [4]
Then, shape of `CircleSlice` would be [?, ?, ?, ?].
This can be represented as following NodeShape.
```cpp
  loco::TensorShape output_shape;
  output_shape.rank(4);
  output_shape.dim(0).unset();
  output_shape.dim(1).unset();
  output_shape.dim(2).unset();
  output_shape.dim(3).unset();
```

In conclusion, current PR can be updated as following.
```suggestion
    loco::TensorShape output_shape;
    output_shape.rank(input_shape.rank());
    for (uint32_t i = 0;i < output_shape.rank(); ++i)
      output_shape.dim(i).unset();
    return output_shape;
```

Reply:
FYI, this is why `Slice` in [initial model](https://github.com/Samsung/ONE/issues/8217#issuecomment-1011694606) has [1, 1, 1] :D
If we dump this with `tfldump`, `shape_signature` may have `[-1, -1, -1]`!

Reply:
```
    for (uint32_t i = 0;i < output_shape.rank(); ++i)
      output_shape.dim(i).unset();
    return output_shape;
```
This will remove current shape info read from the file.
I don't think we have any other nodes do this. Or is there?
Using ` return use_own(node);` was to preserve what was from the file.
Can you explain more about why unset dim values?

Reply:
> If we dump this with tfldump, shape_signature may have [-1, -1, -1]!

I'll check this

Reply:
Your are right!
```
T(2:35) FLOAT32 (1, 1, 1) (-1, -1, -1) B(67) LSTM_2df696de/rnn/while/TensorArrayV2Write/TensorListSetItem6
T(2:38) FLOAT32 (1, 1, 1) (-1, -1, -1) B(70) LSTM_2df696de/rnn/while/TensorArrayV2Write/TensorListSetItem9
```

Reply:
> I don't think we have any other nodes do this. Or is there?

I had a plan with #5683 but nothing done yet because it was not that urgent and no real use case...

> Can you explain more about why unset dim values?

`unset()` makes the dimension `unknown`.
i.e., `output_shape.dim(0).unset()` means `[?]`
https://github.com/Samsung/ONE/blob/ca224b4a329382bf146c06b2c312881f238cdfd4/compiler/loco/include/loco/IR/Dimension.h#L59-L63

Reply:
There's a problem: `Optimized graph is invalid` is dropped.
output node shape vs graph output shape differes: `[1,?,?] vs [5,3,20]`

Reply:
Looking the graph but I cannot understand why `concat( [?,?,?], [?,?,?], [1,3,20] ) = [5, 3, 20]`.... -_-;;

I will look more the model with https://github.com/Samsung/ONE/issues/8217#issuecomment-1011860364...